packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -563,8 +563,8 @@ export class Nodegroup extends Resource implements INodegroup {
  * AMI types of different architectures. Make sure AL2 is always the first element, which will be the default
  * AmiType if amiType and launchTemplateSpec are both undefined.
  */
-const arm64AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_ARM_64, NodegroupAmiType.BOTTLEROCKET_ARM_64];
-const x8664AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64, NodegroupAmiType.BOTTLEROCKET_X86_64,
+const arm64AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_ARM_64, NodegroupAmiType.AL2023_ARM_64_STANDARD, NodegroupAmiType.BOTTLEROCKET_ARM_64];
+const x8664AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64, NodegroupAmiType.AL2023_X86_64_STANDARD, NodegroupAmiType.BOTTLEROCKET_X86_64,
```

Comment:
It's the key to the issue, missing definitions for AmiType: `AL2023_[ARM_64|X86_64]_STANDARD`.